src/com/trilead/ssh2/transport/KexState.java

```diff
@@ -23,10 +23,11 @@
 
 	public BigInteger K;
 	public byte[] H;
-	
+
 	public byte[] hostkey;
-	
-	public DhExchange dhx;
```

Comment:
As above, either keep old fields and values, or clone the class, set the structure how you need (with all the fields encapsulated to prevent this mess re-occuring), and migrate any internal usage to your new class.